### PR TITLE
IGNITE-21215 Add Catalog API for renaming a table

### PR DESCRIPTION
**Pull request by sashapolo**
https://issues.apache.org/jira/browse/IGNITE-21215

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerSelfTest.java [hunk 1]
```diff
@@ -2067,6 +2070,71 @@ void testGetIndexesForTableInSortedOrderById() {
...
+
+        int tableId = manager.tables(prevVersion).iterator().next().id();
+
+        CatalogCommand command = RenameTableCommand.builder()
+                .schemaName(SCHEMA_NAME)
+                .tableName(TABLE_NAME)
+                .newTableName(TABLE_NAME_2)
+                .build();
+
+        CompletableFuture<Void> renameFuture = manager.execute(command);
```
Comment (rpuch) [inline modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerSelfTest.java:2087]: Do we need this variable? Could it be just inlined?

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerSelfTest.java [hunk 2]
```diff
@@ -2067,6 +2070,71 @@ void testGetIndexesForTableInSortedOrderById() {
...
+        CatalogTableDescriptor prevDescriptor = manager.table(tableId, prevVersion);
+        CatalogTableDescriptor curDescriptor = manager.table(tableId, curVersion);
+
+        assertThat(prevDescriptor, is(notNullValue()));
+        assertThat(prevDescriptor.name(), is(TABLE_NAME));
+
+        assertThat(curDescriptor, is(notNullValue()));
+        assertThat(curDescriptor.name(), is(TABLE_NAME_2));
+
+        assertThat(curDescriptor.tableVersion(), is(prevDescriptor.tableVersion() + 1));
```
Comment (rpuch) [inline modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerSelfTest.java:2102]: How about also making sure that column definitions have not changed?

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerSelfTest.java (hunk 1 above)
Comment (sashapolo) [reply modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerSelfTest.java]: fixed

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerSelfTest.java (hunk 2 above)
Comment (sashapolo) [reply modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerSelfTest.java:2101]: fixed

**Review by rpuch — APPROVED** [review]

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/RenameTableCommand.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import org.apache.ignite.internal.catalog.CatalogValidationException;
+import org.apache.ignite.internal.catalog.descriptors.CatalogSchemaDescriptor;
+import org.apache.ignite.internal.catalog.descriptors.CatalogTableDescriptor;
+import org.apache.ignite.internal.catalog.storage.RenameTableEntry;
+import org.apache.ignite.internal.catalog.storage.UpdateEntry;
+
+/**
+ * A command that changes the name of a table.
+ */
+public class RenameTableCommand extends AbstractTableCommand {
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/RenameTableCommand.java:37]: Done

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/RenameTableCommandBuilder.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.commands;
+
+/** Builder for {@link RenameTableCommand}s. */
+public interface RenameTableCommandBuilder extends AbstractTableCommandBuilder<RenameTableCommandBuilder> {
+    RenameTableCommandBuilder newTableName(String newTableName);
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/RenameTableCommandBuilder.java:31]: Done

**Review by korlov42 — APPROVED** [review]

**Pull request merged**